src/main/java/SequenceValidator.java

```diff
@@ -5,167 +5,376 @@
+                                } else {
+                                    cc2ci.get(courseCode).add(new Point(uniqueSemesterId, coursesInSemIdx));
+                                }
+                            }
+
+                            // CREDIT COUNT
+
+                            creditCount += Double.parseDouble((String)course.get("credits"));
+                        }
+                        else if (loopCount == 1)
+                        {
+                            // PREREQUISITE
+
+                            boolean prereqsValid = true;
+                            JSONArray failedRequirements = new JSONArray();
+                            // for each orList of prereqs
+                            JSONArray orLists = null;
+                            try { orLists = course.getJSONObject("requirements").getJSONArray("prereqs");
+                            } catch(Exception e) { continue; /* skip prereq validation if no prerqs */ }
+
+                            for (int i = 0; i < orLists.length(); i++) {
+                                JSONArray orList = orLists.getJSONArray(i);
+
+                                boolean orListValid = false;
+                                // for each course in orList
+                                for (int j = 0; j < orList.length(); j++) {
+                                    String prereqCourseCode = orList.getString(j);
+                                    // prequnisemid = prereq's unique semester id (of first occurence in sequence)
+                                    Integer prequnisemid = null;
+                                    try { prequnisemid = (int) cc2ci.get(prereqCourseCode).get(0).getX(); }
+                                    catch(Exception e) { }
+
+                                    // if prereq is earlier than current course in course sequence
+                                    if (prequnisemid != null && prequnisemid < uniqueSemesterId) {
+                                        orListValid = true;
+                                    }
+                                }
+                                if (!orListValid) {
+                                    prereqsValid = false;
+                                    failedRequirements.put(orList);
+                                }
+                            }
+
+                            if (!prereqsValid) {
+                                sequenceIsValid = false;
+                                issues
+                                .put(new JSONObject()
+                                    .put("type", ISSUES.PREREQUISITE.toString())
+                                    .put("data", new JSONObject()
+                                        .put("courseCode", courseCode)
+                                        .put("position", positionObject(uniqueSemesterId, coursesInSemIdx))
+                                        .put("requirements", failedRequirements)
```

Comment:
What do you think about renaming this property to `unmetRequirements` or `failedRequirements` in the API spec? (not talking about the java variable `failedRequirements`)

Reply:
I very much agree.  I think that a course info json should have a `requirements` property but the validation message should have a `unmetRequirements` property.  So ya, to reiterate, I like `unmetRequirements`.